pkg/controller/prune.go

```diff
@@ -75,14 +76,44 @@ func pruneNoPrivateIP(c *AppGwIngressController, appGw *n.ApplicationGateway, cb
 	return prunedIngresses
 }
 
+// pruneNoSslCertificate filters ingresses which use appgw-ssl-certificate annotation when AppGw doesn't have annotated ssl certificate installed
+func pruneNoSslCertificate(c *AppGwIngressController, appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, ingressList []*v1beta1.Ingress) []*v1beta1.Ingress {
+	var prunedIngresses []*v1beta1.Ingress
+	set := make(map[string]bool)
+	for _, installedSslCertificate := range *appGw.SslCertificates {
+		set[*installedSslCertificate.Name] = true
+	}
+
+	for _, ingress := range ingressList {
+		annotatedSslCertificate, err := annotations.GetAppGwSslCertificate(ingress)
+		if err != nil && annotations.IsInvalidContent(err) {
+			glog.Errorf("Ingress %s/%s has invalid value for annotation %s", ingress.Namespace, ingress.Name, annotations.AppGwSslCertificate)
+		}
+
+		if _, exists := set[annotatedSslCertificate]; !exists {
+			errorLine := fmt.Sprintf("ignoring Ingress %s/%s as it requires Application Gateway %s to have pre-installed ssl certificate", ingress.Namespace, ingress.Name, c.appGwIdentifier.AppGwName)
```

Comment:
Maybe change this to:
`
ignoring Ingress %s/%s as ssl-certificate '%s' specified by the <replace with annotation key> was not found on Application Gateway '%s'
`